colcon_core/verb/__init__.py

```diff
@@ -124,6 +125,35 @@ def check_and_mark_install_layout(install_base, *, merge_install):
     marker_path.write_text(this_install_layout + '\n')
 
 
+def check_and_mark_root_dir(start_path):
+    """
+    Check the marker file for root workspace, otherwise create it.
+
+    The marker filename is `.root_dir`.
+
+    :param str start_path: The path where verb is invoked
+    :raises RuntimeError: if marker file is found in parent directory
+    """
+    current_path = start_path
+    marker_name = '.root_dir'
```

Comment:
Nitpick: please include `colcon` in the name. How about `.colcon_root`? This name should be defined in a global constant.